geowebcache/core/src/main/java/org/geowebcache/storage/CompositeBlobStore.java

```diff
@@ -0,0 +1,360 @@
+        } catch (StorageException e) {
+            throw Throwables.propagate(e);
+        }
+    }
+
+    @Override
+    public void putLayerMetadata(String layerName, String key, String value) {
+        try {
+            store(layerName).putLayerMetadata(layerName, key, value);
+        } catch (StorageException e) {
+            throw Throwables.propagate(e);
+        }
+    }
+
+    @Override
+    public boolean layerExists(String layerName) {
+        for (LiveStore bs : blobStores.values()) {
+            if (bs.config.isEnabled() && bs.liveInstance.layerExists(layerName)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private BlobStore store(String layerId) throws StorageException {
+
+        LiveStore store = forLayer(layerId);
+        if (!store.config.isEnabled()) {
+            throw new StorageException("Attempted to use a blob store that's disabled: "
+                    + store.config.getId());
+        }
+
+        return store.liveInstance;
+    }
+
+    private LiveStore forLayer(String layerName) throws StorageException {
+        TileLayer layer;
+        try {
+            layer = layers.getTileLayer(layerName);
+        } catch (GeoWebCacheException e) {
+            throw new StorageException("Error getting layer " + layerName, e);
+        }
+        String storeId = layer.getBlobStoreId();
+        LiveStore store;
+        if (null == storeId) {
+            store = defaultStore();
+        } else {
+            store = blobStores.get(storeId);
+        }
+        if (store == null) {
+            throw new StorageException("No BlobStore with id '" + storeId + "' found");
+        }
+        return store;
+    }
+
+    private LiveStore defaultStore() throws StorageException {
+        LiveStore store = blobStores.get(CompositeBlobStore.DEFAULT_STORE_DEFAULT_ID);
+        if (store == null) {
+            throw new StorageException("No default BlobStore has been defined");
+        }
+        return store;
+    }
+
+    /**
+     * Loads the blob stores from the list of configuration objects
+     * 
+     * @param configs the list of blob store configurations
+     * @return a mapping of blob store id to {@link LiveStore} containing the configuration itself
+     *         and the live instance if the blob store is enabled
+     * @throws ConfigurationException if there's a configuration error like a store confing having
+     *         no id, or two store configs having the same id, or more than one store config being
+     *         marked as the default one, or the default store is not
+     *         {@link BlobStoreConfig#isEnabled() enabled}
+     * @throws StorageException if the live {@code BlobStore} instance can't be
+     *         {@link BlobStoreConfig#createInstance() created} of an enabled
+     *         {@link BlobStoreConfig}
+     */
+    Map<String, LiveStore> loadBlobStores(List<? extends BlobStoreConfig> configs)
```

Comment:
It seems to me that `Collection` would be good enough without needing the more specific `List`.